File: leetcode/62. Unique Paths/main.java
// Solved by @kitanoyoru
// https://leetcode.com/problems/unique-paths/

import java.util.*;

class Solution {
  public int uniquePaths(int m, int n) {
    int dp = new int[n];
    Arrays.fill(dp, 0);

    dp[0] = 1;

    for (int i = 0; i < m; i++) {
      for (int j = 0; j < n; j++) {
        dp[j] += dp[j-1];
      }
    }

    return dp[n-1];
  }
}

File: leetcode/62. Unique Paths/main.ts
// Solved by @kitanoyoru
// https://leetcode.com/problems/unique-paths/

const uniquePaths = (m: number, n: number) => {
  if (m == 1 || n == 1) {
    return 1
  }

  let dp = new Array(n).fill(0)
  dp[0] = 1

  for (let row = 0; row < m; row++) {
    for (let col = 1; col < n; col++) {
      dp[col] += dp[col - 1]
    }
  }

  return dp[n - 1]
}
